### PR TITLE
add: scaffolder

### PR DESCRIPTION
**Pull request by ankitjena**
#24 Create generated schema and resolvers file format.

### CODE REVIEW

File: src/GraphQLBackend.ts [hunk 1]
```diff
@@ -72,6 +77,10 @@ export class GraphQLBackendCreator {
     this.dbContextProvider = provider;
   }
 
+  public setOutputWriter(output: OutputWriter) {
```
Comment (wtrocki) [inline src/GraphQLBackend.ts]: I will go with config and have file as default

File: src/Scaffold.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { IGraphQLBackend } from './GraphQLBackend'
+import { writeFile, mkdir } from 'fs'
+import { join } from 'path'
+import { promisify } from 'util'
+
+const writeFileAsync = promisify(writeFile)
+const mkdirAsync = promisify(mkdir)
+
+export interface OutputWriter {
```
Comment (wtrocki) [inline src/Scaffold.ts]: missing comments

File: src/Scaffold.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  generateProjectStructure(backend)
+}
+
+export class ProjectScaffolder implements OutputWriter {
+
+  public async generateProjectStructure(backend: IGraphQLBackend) {
+    try {
+      let outputPath = `${process.cwd()}/generated`
+      const resolvers = JSON.stringify(backend.resolvers, undefined, 2)
```
Comment (wtrocki) [inline src/Scaffold.ts]: That probably needs to be handlebars template rather than json

File: src/Scaffold.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      await mkdirAsync(outputPath)
+      await writeFileAsync(join(outputPath, 'schema.graphql'), backend.schema)
+      await writeFileAsync(join(outputPath, 'resolvers.ts'), resolvers)
+    } catch(e) {
+      console.log(e)
```
Comment (wtrocki) [inline src/Scaffold.ts]: console.log. Maybe we can get some logger abstraction later

**Comment by wtrocki** [comment]
Set method is not caled

**Comment by ankitjena** [comment]
> Set method is not caled

@wtrocki The above methods like registerResolverTypes and setDatabaseContext aren't called any where either.

File: src/GraphQLBackend.ts (hunk 1 above)
Comment (ankitjena) [reply src/GraphQLBackend.ts]: Didn't understand this?

**Comment by wtrocki** [comment]
They are part of end user api and do have defaults set anyway. In this case we do not set any value or defaults and have this object called in main function. You are probably getting undefined is not a function or something like thay because of it.

**Comment by wtrocki** [comment]
@ankitjena What happened ?

**Pull request closed without merging**